lib/ansible/constants.py

```diff
@@ -56,8 +51,31 @@ def __getitem__(self, y):
         return self._value[y]
 
 
+def __getattr__(config_constant):
+    ''' Handle dynamicall generating a 'constant' when first requested,
+        otherwise just return it from cached value.
+    '''
+
+    if config_constant not in globals():
+        try:
+            value = config.get_config_value(config_constant, variables=globals())
+        except Exception as e:
+            raise AttributeError(e)
```

Comment:
Don't lose the context, set an explicit cause:
```suggestion
            raise AttributeError(e) from e
```